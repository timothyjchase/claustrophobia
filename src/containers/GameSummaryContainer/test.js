import React from 'react'
import { GameSummaryContainer, mapStateToProps } from './index'

describe('<GameSummaryContainer />', () => {
  describe('with scenarioKey = TEST_SCENARIO', () => {
    const props = {
      currentState: {
        scenarioKey: 'TEST_SCENARIO',
        turn: 1,
        demonDice: 2,
        threatDice: 3,
        demonsInPlay: 4,
        toughTrogsInPlay: 5,
        trogsInPlay: 6,
      },
      removeDemon: jest.fn(),
      removeToughTrog: jest.fn(),
      removeTrog: jest.fn(),
    }
    const component = shallow(<GameSummaryContainer {...props} />)

    it('should match snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })
  describe('with scenarioKey = null', () => {
    const props = {
      currentState: {
        scenarioKey: null,
        turn: 1,
        demonDice: 2,
        threatDice: 3,
        demonsInPlay: 4,
        toughTrogsInPlay: 5,
        trogsInPlay: 6,
      },
      removeDemon: jest.fn(),
      removeToughTrog: jest.fn(),
      removeTrog: jest.fn(),
    }
    const component = shallow(<GameSummaryContainer {...props} />)

    it('should match snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })
})

describe('mapStateToProps', () => {
  describe('with basic state', () => {
    const state = { current: {} }
    it('should return currentState', () => {
      expect(mapStateToProps(state)).toEqual({ currentState: {} })
    })
  })
})
