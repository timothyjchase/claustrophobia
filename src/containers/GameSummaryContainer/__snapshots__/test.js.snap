// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GameSummaryContainer /> with scenarioKey = TEST_SCENARIO should match snapshot 1`] = `
<GameSummary
  demonDice={2}
  threatDice={3}
  turn={1}
  warriors={
    Array [
      Object {
        "name": "Test Demon",
        "numberInPlay": 4,
        "onRemove": [MockFunction],
      },
      Object {
        "name": "Tough Troglodyte",
        "numberInPlay": 5,
        "onRemove": [MockFunction],
      },
      Object {
        "name": "Troglodyte",
        "numberInPlay": 6,
        "onRemove": [MockFunction],
      },
    ]
  }
/>
`;

exports[`<GameSummaryContainer /> with scenarioKey = null should match snapshot 1`] = `
<GameSummary
  demonDice={2}
  threatDice={3}
  turn={1}
  warriors={
    Array [
      Object {
        "numberInPlay": 4,
        "onRemove": [MockFunction],
      },
      Object {
        "name": "Tough Troglodyte",
        "numberInPlay": 5,
        "onRemove": [MockFunction],
      },
      Object {
        "name": "Troglodyte",
        "numberInPlay": 6,
        "onRemove": [MockFunction],
      },
    ]
  }
/>
`;
